test/rpc_test.py

```diff
@@ -533,6 +533,52 @@ def test_py_user_defined(self):
         )
         self.assertEqual(ret, my_function(n, n + 1, n + 2))
 
+    def _profiler_test_with_rpc(self, rpc_exec_mode, func, args):
+        dst = (self.rank + 1) % self.world_size
+        # only run profiler on rank 0.
+        if self.rank in {0}:
+            with torch.autograd.profiler.profile() as prof:
+                if rpc_exec_mode == RPCExecMode.SYNC:
+                    rpc.rpc_sync("worker{}".format(dst), func, args=args)
+                elif rpc_exec_mode == RPCExecMode.ASYNC:
+                    fut = rpc.rpc_async("worker{}".format(dst), func, args=args)
+                    fut.wait()
+                else:
+                    self.assertTrue(rpc_exec_mode == RPCExecMode.REMOTE)
+                    rref = rpc.remote("worker{}".format(dst), func, args=args)
+                    rref.to_here()
+                    # Hack: We need to wait for the instance to be created on
+                    # the owner, and get back a positive confirmation.
+                    # Calling to_here does not ensure that we have finished
+                    # processing the Owner's confirmation of this RRef.
+                    time.sleep(1)
```

Comment:
`to_here` should wait for the owner to confirm the RRef. Could you describe what sort of issues you encountered without this sleep?

Reply:
After a quick look, I'm not completely sure where `to_here()` waits for the owner to confirm the Rref. It looks like `toHere()` creates a `MessageType::PYTHON_RREF_FETCH_CALL` request, and doesn't block on the owner having sent a confirmation (`REMOTE_RET` msg type). 

The issue I encountered was that blocking with `to_here()` sometimes does not result in the original future (the one that sends the `PYTHON_REMOTE_CALL`) that sent the request to create the RRef being satisfied. Then the autograd profiler would exit and disable the profiler, but the `RecordFunction` attached to that future would not have invoked its callbacks, so the result would not be present in the profiler.